cli/src/pcluster/validators/networking_validators.py

```diff
@@ -69,12 +108,17 @@ def _validate(self, elastic_ip: Union[str, bool]):
 
 
 class SingleSubnetValidator(Validator):
-    """Validate all queues reference the same subnet."""
+    """Validate only one subnet is used for compute resources with single instance type."""
 
-    def _validate(self, queues_subnets):
-        subnet_ids = {tuple(set(queue_subnets)) for queue_subnets in queues_subnets}
+    def _validate(self, queue_name, subnet_ids):
```

Comment:
A validator should be able to determine the correct/incorrect state using only the information it is provided.
This validator already assumes that the queue is being used with a Single Instance Type (since it's only registered when there is a Single Instance Type usage).

I think validation logic should be local to the validator. The list of ComputeResources can be passed to this validator and it should then determine what's valid/invalid.

Reply:
This will also allow the validator to communicate which Compute Resource is invalid.

Reply:
I had to do this to avoid a circular import between `cluster_config.py` and `networking_validators.py`. We can discuss about this.